### PR TITLE
refactor Multi-cluster GitOps pattern

### PR DESCRIPTION
**Pull request by tsahiduek**
*Issue #, if available:*

*Description of changes:*
Refactor the entire pattern, while keeping the structure as it is (3 environments, 3 teams, GitOps bootstrap for multi-software-tenant delivery.

Changes:
* Removing unnecessary addons - keeping minimal required addons for demonstration purpose
* Moving from MNG to Karpenter with Fargate profile only for Karpenter
* Simplifying code with new helper functions 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by tsahiduek** [comment]
updated pattern to use new `withEnv` function in version 10.0.1

File: lib/pipeline-multi-env-gitops/index.ts [hunk 1]
```diff
@@ -2,270 +2,269 @@ import * as blueprints from '@aws-quickstart/eks-blueprints';
...
+                },
+            },
         });
 
+        // commonly configured addons
+        const addons: blueprints.ClusterAddOn[] = [
+            new blueprints.AwsLoadBalancerControllerAddOn(),
+            new blueprints.CertManagerAddOn(),
+            new blueprints.SecretsStoreAddOn({
+                values: {
```
Comment (shapirov103) [inline lib/pipeline-multi-env-gitops/index.ts:84]: Minor: this config is way too nested. You could use something like [dot-object.](https://www.npmjs.com/package/dot-object)  to deal with nested json. 

no action needed, just a suggestion for future as it provides marginal improvement

File: lib/pipeline-multi-env-gitops/index.ts [hunk 2]
```diff
@@ -2,270 +2,269 @@ import * as blueprints from '@aws-quickstart/eks-blueprints';
...
+                    linux: {
+                        affinity: {
+                            nodeAffinity: {
+                                requiredDuringSchedulingIgnoredDuringExecution:
+                                    {
+                                        nodeSelectorTerms: [
+                                            {
+                                                matchExpressions: [
+                                                    {
+                                                        key: 'eks.amazonaws.com/compute-type',
```
Comment (shapirov103) [inline lib/pipeline-multi-env-gitops/index.ts:94]: why do we need to set affinity outside of fargate in this case? Aren't we using fargate only for karpenter namespace?

**Review by shapirov103 — COMMENTED** [review]
@tsahiduek nice one!  A couple of mostly minor comments.

**Comment by shapirov103** [comment]
Validated functionally in my accounts, no issues.

File: package.json [hunk 3]
```diff
@@ -41,7 +41,7 @@
     "@instana/aws-eks-blueprint-addon": "^1.0.4"
   },
   "overrides": {
-    "@aws-quickstart/eks-blueprints": "1.10.0",
+    "@aws-quickstart/eks-blueprints": "1.10.1",
     "aws-cdk": "2.86.0",
```
[the remark this answers is not in the document]
Comment (tsahiduek) [reply package.json]: Done.

**Comment by tsahiduek** [comment]
Hi @shapirov103 , I think I've addressed all of the above comments. 
Please lmk if we can merge it.
Thank you

**Pull request merged**